modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java

```diff
@@ -289,15 +298,20 @@ public static IgniteState state(@Nullable String name) {
      * Stops default grid. This method is identical to {@code G.stop(null, cancel)} apply.
      * Note that method does not wait for all tasks to be completed.
      *
-     * @param cancel If {@code true} then all jobs currently executing on
-     *      default grid will be cancelled by calling {@link ComputeJob#cancel()}
-     *      method. Note that just like with {@link Thread#interrupt()}, it is
-     *      up to the actual job to exit from execution
+     * @param cancel If {@code true} then all jobs currently will be cancelled
+     *      by calling {@link ComputeJob#cancel()} method. Note that just like with
+     *      {@link Thread#interrupt()}, it is up to the actual job to exit from
+     *      execution. If {@code false}, then jobs currently running will not be
+     *      canceled. In either case, grid node will wait for completion of all
+     *      jobs running on it before stopping. If {@code null} then
+     *      not-{@link IgniteConfiguration#isGracefulShutdown()} is used instead.
+     * @param waitForBackups If {@code true} then node will wait until all of its data is backed up before shutting
+     *      down. If {@code null} then {@link IgniteConfiguration#isWaitForBackupsOnShutdown()} is used instead.
```

Comment:
I don't like an idea of considering null as IgniteConfiguration#isWaitForBackupsOnShutdown(). Is it possible to use `IgniteSystemProperties.getBoolean(IgniteSystemProperties.IGNITE_WAIT_FOR_BACKUPS_ON_SHUTDOWN)` instead of null within methods like `org.apache.ignite.Ignition#stop(boolean)`?

Reply:
@sanpwc This is private API so I think it is tolerable.
If we require that IgniteSystemProperties.IGNITE_WAIT_FOR_BACKUPS_ON_SHUTDOWN is supplied explicitly by caller, this means that e.g. FailureHandler implementations have to compute these properties' values.
Ignition API will be clear without nulls when we add it in Phase II. WDYT?